### ISSUE TITLE
Add learning_rate and batch_size to saving and loading. 

### ISSUE THREAD

**Issue by xinsuinizhuan** [josephjaspers/blackcat_tensors#32]
when i save the network，then load the network, run it, it error，it need to reset learning_rate and batch_size。I sugest  to save the  learning_rate and batch_size when save the network, and load them, do not need to reset them.

**Comment by josephjaspers** [comment]
Yes that sounds like a good feature. I will add it when I can.

**Comment by xinsuinizhuan** [comment]
when i load and save model by save and load function, compile error:
![1569575028(1)](https://user-images.githubusercontent.com/40679769/65756999-ce818380-e148-11e9-901f-067fcc9e50a9.jpg)

**Comment by josephjaspers** [comment]
The make_lstm_network should be....

```
auto make_lstm_network() {
return BC::nn::neuralnetwork(

    BC::nn::lstm(BC::host_tag(), 96*10, 128),
    BC::nn::recurrent(BC::host_tag(),  128, 96),
    BC::nn::softmax(BC::host_tag(), 96),

)

}

**Comment by josephjaspers** [comment]
Added as of:
https://github.com/josephjaspers/blackcat_tensors/commit/5ce395b5b44d64153adde83d3033946c5990bd33

**Comment by josephjaspers** [comment]
https://github.com/josephjaspers/blackcat_tensors/commit/582203819850d5f1df769a46bcea389edcee8690

**Comment by xinsuinizhuan** [comment]
Oh，i see. It is my error! Thank you!
